### PR TITLE
MULE-19139: Mapped Error Type gets replaced by MULE:UNKNOWN

### PR DESCRIPTION
**Pull request by IvanAndresFritzler**

### CODE REVIEW

**Review by elrodro83 — DISMISSED** [review]

**Comment by IvanAndresFritzler** [comment]
--validate

File: src/test/munit/all-validation-test-suite.xml [hunk 1]
```diff
@@ -41,4 +41,28 @@
         </munit:validation>
     </munit:test>
 
+    <munit:test name="validationAllWithErrorMappingResolvingToMuleUnknown"
+                description=""
+                tags="MULE-19139"
+                ignore="#[Munit::muleVersionPriorTo('4.3.1')]">
```
Comment (elrodro83) [inline src/test/munit/all-validation-test-suite.xml:47]: this should run properly on 4.2.x as well

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: src/test/munit/all-validation-test-suite.xml [hunk 2]
```diff
@@ -41,4 +40,28 @@
         </munit:validation>
     </munit:test>
 
+    <munit:test name="validationAllWithErrorMappingResolvingToMuleUnknown"
+                description=""
+                tags="MULE-19139"
+                ignore="#[Munit::muleVersionPriorTo('4.2.2') or Munit::muleVersionEqualTo('4.2.2')]">
```
Comment (elrodro83) [inline src/test/munit/all-validation-test-suite.xml:46]: why not just `#[Munit::muleVersionPriorTo('4.2.3')]`

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: src/test/munit/all-validation-test-suite.xml (hunk 2 above)
Comment (IvanAndresFritzler) [reply src/test/munit/all-validation-test-suite.xml]: 4.2.3 would leave out 4.2.3-SNAPSHOT, so I changed it to 4.2.3-SNAPSHOT

**Review by elrodro83 — APPROVED** [review]

**Pull request merged**